WasteWar/Assets/Scripts/Drawing/StructurePlacement.cs

```diff
@@ -0,0 +1,102 @@
+        failedConditions = hitLocation.x > terrainSize.z ? (failedConditions + 1) : failedConditions;
+        failedConditions = hitLocation.x < 0 ? (failedConditions + 1) : failedConditions;
+        failedConditions = hitLocation.z > terrainSize.z ? (failedConditions + 1) : failedConditions;
+        failedConditions = hitLocation.z < 0 ? (failedConditions + 1) : failedConditions;
+
+        return failedConditions == 0;
+    }
+    private void PlaceStructureOnGridOnClick()
+    {
+        bool isSpaceOccupied=false;
+
+    //will create some kind of switch case for this later on
+        if (Input.GetKey(KeyCode.B))
+        {
+            Destroy(buildingTemplate);
+            buildingTemplate = Instantiate(buildingPrefab1, ObjectSnapper.SnapToGridCell(hit.point, GridConstants.Instance.FloatCellSize()), Quaternion.Euler(0, 0, 0));
+            isAStructureSelected = true;
+        }
+        else if (Input.GetKey(KeyCode.C))
+        {
+            Destroy(buildingTemplate);
+            buildingTemplate = Instantiate(buildingPrefab2, ObjectSnapper.SnapToGridCell(hit.point, GridConstants.Instance.FloatCellSize()), Quaternion.Euler(0, 0, 0));
+            isAStructureSelected = true;
+        }
+        else if (Input.GetKey(KeyCode.Escape))
+            {
+                Destroy(buildingTemplate);
+                isAStructureSelected = false;
+            }
+        if (isAStructureSelected == true)
+        {
+            isSpaceOccupied = structureGrid.IsGridCellFilled(hit.point, buildingTemplate.GetComponent<Renderer>().bounds.size);
+           
+            if (!isSpaceOccupied)
+                buildingTemplate.GetComponent<MeshRenderer>().material.SetColor("_Color", Color.green);
+            else
+            {
+                buildingTemplate.GetComponent<MeshRenderer>().material.SetColor("_Color", Color.red);
+            }
+            buildingTemplate.transform.position = ObjectSnapper.SnapToGridCell(hit.point, GridConstants.Instance.FloatCellSize(), buildingTemplate.GetComponent<Renderer>().bounds.size);
+        }
+
+        if (Input.GetMouseButtonDown(0) && isAStructureSelected == true && !isSpaceOccupied)
+        {
+            Destroy(buildingTemplate);
+            isAStructureSelected = false;
+            structureGrid.AddStructure(hit.point, buildingTemplate.GetComponent<Renderer>().bounds.size);
+            DrawStructureOnGrid(Structures,buildingTemplate,hit.point);
+            Debug.Log(hit.point);
+        }
+    }
+
+    private static void DrawStructureOnGrid(List<GameObject> Structures,GameObject structureToAdd,Vector3 coords)
+    {
+        structureToAdd.GetComponent<MeshRenderer>().material.SetColor("_Color", Color.white);
+        Structures.Add(Instantiate(structureToAdd, ObjectSnapper.SnapToGridCell(
+            coords, 
+            GridConstants.Instance.FloatCellSize(), structureToAdd.GetComponent<Renderer>().bounds.size), 
+            Quaternion.Euler(0, 0, 0)
```

Comment:
I assume I keep seeing this because that's a default rotation, if so serialize this so we can modify it while playing